charon/config.py

```diff
@@ -39,6 +40,15 @@ def __init__(self, data: Dict):
         self.__ignore_signature_suffix: Dict = data.get("ignore_signature_suffix", None)
         self.__signature_command: str = data.get("detach_signature_command", None)
         self.__aws_cf_enable: bool = data.get("aws_cf_enable", False)
+        self.__amqp_queue: str = data.get("amqp_queue", None)
+        self.__sign_result_loc: str = data.get("sign_result_loc", None)
```

Comment:
@yma96 I think this one is also needed to be passed by cmd flag. The cause is that we should store this in Konflux workspace storage, but the path to this workspace is a variable which is passed by Konflux for each release pipeline run. That means it will always change and we don't know the exact value. So the only way here is give it through flag and then we can specify it through Konflux shell environment.

Reply:
@ligangty, ok, make sense.

Reply:
BTW, can I ask this "sign_result_loc" 's  purpose? Is it used to store the sign result bundle or extracted signing files?

Reply:
@ligangty, this's used to store the oras pull result json file from radas quay registry, and let maven-publish could detect this file to handle the following .asc files generation, the final singing files will be generated directly alongside the artifact.

Reply:
@yma96 Oh ok. Where will you want to process this file in charon? During the maven uploading phase?

Reply:
@ligangty, this "sign_result_loc" should only be used for "sign"(when message received), oras pulls the files result to this loc and store the files result into the SignHandler:_downloaded_files, and we could use classmethod get_downloaded_files to detect the files result during "process"(when maven upload), since this kind of class method is bound to the class and not the instance of the class, so that could be shared by all instances of the class. Through this python feature, we may not need to transfer this value across two commands.

Reply:
@yma96 one big issue is "sign" and "maven_upload" are not happend in same time. In later Konflux usage, they are called in different tasks with two command call. That means we could not use any memory way (like you said for the classmethod) to store these files. We should store them into a Konflux reachable storage between different tasks. That's why I proposed to use a flag to pass this file location.

Reply:
@ligangty Ok, I just think of a flag to detect, not aware of the sign is a dependent task, so the SignHandler status management here should not be workable, and the corresponding process status handling should be adjusted too.

How about environ SIGN_RESULT_LOC for sign_result_loc after sign msg received? For maven upload, could it be able to fetch the env SIGN_RESULT_LOC across different tasks?

Or just open this flag for both of the two tasks, and Konflux could help to transfer that sequentially?

Reply:
@yma96 the env var is not shared between two tekton tasks. That's why the task is using the "result"(see https://medium.com/@email2smohanty/passing-values-between-tekton-tasks-using-task-results-9ac89a44b2b9). What I'm thinking here is passing this sign_result_loc into sign command in sign task, and then store it in to task result. Then extract it in the maven upload task by passing it to maven upload command.

Reply:
@ligangty, make sense, so both of the two tasks would open this loc as the cmd flag IIUC.